nucypher/blockchain/eth/deployers.py

```diff
@@ -855,3 +873,66 @@ def deploy(self, secret_hash: bytes, gas_limit: int = None, progress=None) -> Di
         self._contract = adjudicator_contract
 
         return deployment_receipts
+
+
+class WorkLockDeployer(BaseContractDeployer):
+
+    agency = WorkLockAgent
+    contract_name = agency.registry_contract_name
+    deployment_steps = ('contract_deployment', 'set-worklock')
+    _upgradeable = False
+    __proxy_deployer = NotImplemented
+
+    def __init__(self, *args, **kwargs):
+
+        super().__init__(*args, **kwargs)
+        self.staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=self.registry)
+        self.token_agent =  ContractAgency.get_agent(NucypherTokenAgent, registry=self.registry)
+
+    def deploy(self, gas_limit: int = None, progress = None) -> Dict[str, str]:
+        """
+
+        Worklock Constructor Parameters (Ordered)
+        ============================================================================
+        _token Token contract
+        _escrow Escrow contract
+        _startBidDate Timestamp when bidding starts
+        _endBidDate Timestamp when bidding will end
+        _depositRate ETH -> NU rate
+        _refundRate Work -> ETH rate
+        _lockedPeriods Number of periods during which claimed tokens will be locked
+
+        """
+        self.check_deployment_readiness()
+
+        # Deploy
+        constructor_args = (self.token_agent.contract_address,
+                            self.staking_agent.contract_address,
+                            *self.economics.worklock_deployment_parameters)
+
+        worklock_contract, deploy_txhash = self.blockchain.deploy_contract(self.deployer_address,
+                                                                           self.registry,
+                                                                           self.contract_name,
+                                                                           *constructor_args,
+                                                                           gas_limit=gas_limit)
+        if progress:
+            progress.update(1)
+
+        bonding_function = self.staking_agent.contract.functions.setWorkLock(worklock_contract.address)
+        bonding_receipt = self.blockchain.send_transaction(sender_address=self.deployer_address,
+                                                           contract_function=bonding_function)
+        if progress:
+            progress.update(1)
+
+        # Gather the transaction hashes
+        self.deployment_transactions = {'deployment': deploy_txhash, 'bond_escrow': bonding_receipt}
+        self._contract = worklock_contract
+        return self.deployment_transactions
+
+    def fund(self, sender_address: str) -> dict:
```

Comment:
Actually, a good question if it's ok to fund from a physical account. I guess, it is though since it's the beginning of the network anyway